distributed/tests/test_worker.py

```diff
@@ -1634,3 +1634,22 @@ def bad_heartbeat_worker(*args, **kwargs):
                 else:
                     assert w.status == "closed"
     assert "Heartbeat to scheduler failed" in logger.getvalue()
+
+
+@gen_cluster(client=True, nthreads=[("127.0.0.1", 1)])
+async def test_delete_data(c, s, w):
+    df = dask.datasets.timeseries(dtypes={"x": int, "y": float}, freq="1s").reset_index(
+        drop=True
+    )
```

Comment:
Do we need a dataset this complex?  Do we need these datatypes or the reset_index call?  If not, lets remove them to keep things as simple as possible for future reviewers.

Almost all tests in this file intentionally use very simple computations, like `future = client.submit(inc, 1)`.  I recommend that we do the same here unless having a dataframe with a reset index is critical to test the delete_data operation.

Reply:
Actually, no. I can simplify that a bit.